tcosconfig.1
.\"Created with GNOME Manpages Editor Wizard
.\"http://gmanedit.sourceforge.net
.\"Sergio Rua <srua@gpul.org>
.\"
.TH tcosconfig 1 "Mar 11, 2007" "tcosconfig man page"

.SH NAME
tcosconfig \- TCOS tool to create and personalize initramfs network boot files

.SH DESCRIPTION

.PP
\fBtcosconfig\fP is a python + gtk2 gui for gentcos command


.SH OPTIONS
.B [no options]
 With no options will exec TcosConfig in normal mode

.B --debug
 Show verbose output when running

.B --fakearch=foo
 Don't detect architecture, force foo.

.B --help
 Some usage help about tcosconfig


.SH BUGS
Contact Mario Izquierdo <mariodebian@gmail.com>

Homepage: http://www.tcosproject.org

.SH SEE ALSO
gentcos(1)

.SH AUTHOR
Mario Izquierdo (mariodebian) <mariodebian@gmail.com>


Changelog
tcosconfig (0.0.10) unstable; urgency=low

  * Add more params (wireless, rdesktop, ...)
  * Close all expanders by default
  * When open one expander close others

 -- Mario Izquierdo (mariodebian) <mariodebian@gmail.com>  Wed, 14 Feb 2007 18:16:44 +0100

tcosconfig (0.0.9) unstable; urgency=low

  * Change /bin/sh to /bin/bash (solve ubuntu dash problem)
  * Remove default suffix "-tcos1" to empty

 -- Mario Izquierdo (mariodebian) <mariodebian@gmail.com>  Sun, 21 Jan 2007 14:04:04 +0100

tcosconfig (0.0.8) unstable; urgency=low

  * Redesign GUI with expanders.
  * Add some new vars to gui: 
      TCOS_INSTALLER, TCOS_DEBOOTSTRAP, TCOS_XORG_OPENGL TCOS_XORG_ALLDRIVERS
      TCOS_SOUND_ISA, TCOS_MIN_RAM, TCOS_PULSEAUDIO_RESAMPLE_METHOD

 -- Mario Izquierdo (mariodebian) <mariodebian@gmail.com>  Sat, 16 Dec 2006 21:53:26 +0100

tcosconfig (0.0.7) unstable; urgency=low

  * Rename source package tcos-config => tcosconfig

 -- Mario Izquierdo (mariodebian) <mariodebian@gmail.com>  Sat, 28 Oct 2006 22:55:11 +0200

tcos-config (0.0.6) unstable; urgency=low

  * Add TCOS_USENFS(ComboBox) and TCOS_PULSEAUDIO(CheckBox)
  * Change *.py header enconding is now UTF-8 and not UTF8, xgettext don't work with UTF8

 -- Mario Izquierdo (mariodebian) <mariodebian@gmail.com>  Thu, 31 Aug 2006 11:39:15 +0200

tcos-config (0.0.5) unstable; urgency=low

  * New stable release.
  * Add about dialog and logo (tcos_config.png)
  * fix-glade.sh script is broken, need to edita glade by hand.

 -- Mario Izquierdo (mariodebian) <mariodebian@gmail.com>  Fri,  7 Jul 2006 16:25:28 +0200

tcos-config (0.0.4-b3) unstable; urgency=low

  * Change GtkComboBox glade values with bidimensional list in shared file
    this allow to translate strings ans simplify upgrades.
  * Clean python code (remove TcosConfig unnecesary class)
  * Begin to document some functions
  * Fix ConfigReader::changevalue() reset after save to reload settings
  * Remove unnecesary modules
  * Fix some dirs problems (exec in sources based on tcosconfig.py file exists and not in .svn dir)
  * Change tcosconfig.sh script.

 -- Mario Izquierdo (mariodebian) <mariodebian@gmail.com>  Fri,  7 Jul 2006 15:37:32 +0200

tcos-config (0.0.4-b2) unstable; urgency=low

  * Updated dependencies to python2.4-gtk and python2.4-glade

 -- Mario Izquierdo (mariodebian) <mariodebian@gmail.com>  Wed, 31 May 2006 11:21:06 +0200

tcos-config (0.0.4-b1) unstable; urgency=low

  * Add TCOS_REMOTEFS selector
  * Add TCOS_ADMIN_USER entry
  * Add TCOS_SHOW_NEEDED MODULES checkbox
  * Delete TCOS_BIG_BUSYBOX checkbox
  * Delete TCOS_SSH_KEYS filechooser

 -- Mario Izquierdo (mariodebian) <mariodebian@gmail.com>  Tue, 30 May 2006 16:29:19 +0200

tcos-config (0.0.4) unstable; urgency=low

  * Fixed BUG tcos.conf duplicate entries, need clean self.newvars before save it
  * TCOS_KERNEL now works.
  * Add new vars ex: TCOS_X11VNC checkbox

 -- Mario Izquierdo (mariodebian) <mariodebian@gmail.com>  Thu, 25 May 2006 20:48:33 +0200

tcos-config (0.0.3) unstable; urgency=low

  * First usable release.
  * Save config works.
  * TCOS_KERNEL and TCOS_SSH_KEYS not supported yet

 -- Mario Izquierdo (mariodebian) <mariodebian@gmail.com>  Sun, 14 May 2006 20:04:19 +0200

tcos-config (0.0.2) unstable; urgency=low

  * New improvements.
  * GUI based on guadalinex/ubuntu-express installer (glade file)
  * Split python in some classes.

 -- Mario Izquierdo (mariodebian) <mariodebian@gmail.com>  Fri, 12 May 2006 23:38:44 +0200

tcos-config (0.0.1) unstable; urgency=low

  * Initial release.

 -- Mario Izquierdo (mariodebian) <mariodebian@gmail.com>  Wed, 19 Apr 2006 09:22:47 +0200


tcosconfig.py
#!/usr/bin/env python2.4
# -*- coding: UTF-8 -*-
##########################################################################
# tcos_config writen by MarioDebian <mariodebian@gmail.com>
#
#    TcosConfig version __VERSION__
#
# Copyright (c) 2005 Mario Izquierdo <mariodebian@gmail.com>
#
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2, or (at your option)
# any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program; if not, write to the Free Software
# Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
# 02111-1307, USA.
###########################################################################



# debug is false, you can active by tcosconfig --debug
# default config shared class (shared)
# http://mail.python.org/pipermail/tutor/2002-November/018353.html
import shared



def print_debug(txt):
    if shared.debug:
        print txt
    return
    
import sys
import getopt

from DetectArch import DetectArch
fakearch=None


def usage():
    print "TcosConfig help:"
    print ""
    print "   tcosconfig -d [--debug]  (write debug data to stdout)"
    print "   tcosconfig -h [--help]   (this help)"

    
try:
    opts, args = getopt.getopt(sys.argv[1:], ":hd", ["help", "debug", "fakearch="])
except getopt.error, msg:
    print msg
    print "for command line options use tcosconfig --help"
    sys.exit(2)


# process options
for o, a in opts:
    if o in ("-d", "--debug"):
        print "DEBUG ACTIVE"
        shared.debug = True
    if o in ("-h", "--help"):
        usage()
        sys.exit()
    if o == "--fakearch":
        fakearch=a


arch=DetectArch()
serverarch=arch.get(fakearch)

if serverarch != "i386":
    if arch.buildChroot():
        if not shared.updatetcosimages:
            sys.exit(0)



    
# my own classes
from TcosGui import *



gui = TcosGui()
# Run app
gui.run()

    
